src/js/components/TaskView.js

```diff
@@ -219,11 +241,7 @@ class TaskView extends mixin(SaveStateMixin, StoreMixin) {
       });
     });
 
-    if (searchString !== '') {
-      tasks = StringUtil.filterByString(tasks, 'id', searchString);
-    }
-
-    tasks = this.filterByCurrentStatus(tasks);
+    tasks = this.getFilteredTasks(filterByStatus, tasks, searchString);
```

Comment:
We shouldn't pass around props and state as arguments :) Lets take out all the arguments in these methods that are accessible by state or props!